packages/vite/src/node/plugins/html.ts

```diff
@@ -418,6 +418,7 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
         }
 
         await traverseHtml(html, id, (node) => {
+          id = normalizePath(id)
```

Comment:
We should move this out of the loop so it isn't called for every node. Normally, the id in `transform(code, id)` should also already be normalized. Maybe we aren't doing it in our custom handling of html? We can patch the issue and then review that too.

Reply:
I debugged it and found that the id in `transform(html, id)` was not normalized. So we should move normalization to?
https://github.com/vitejs/vite/blob/2836276dc0e132c4d64dc4c08fc730058406f2c0/packages/vite/src/node/plugins/html.ts#L325

Reply:
Shouldn't `path.posix.relative` already return a normalized path? I mean checking why the `id` passed to `transform(html, id)` isn't normalized. I don't know if it is an easy fix, so I'm good moving the normalizePath out of `traverseHtml` for now.

Reply:
You seem to have misunderstood me. What I want to confirm is whether your previous description wants me to move the modification of line 421 to line 324. 😄

Reply:
Ah, yes! :)
You can then reuse `normalizedId` for the call at 325 👍🏼